packages/clay-date-picker/src/index.tsx

```diff
@@ -227,18 +240,16 @@ const ClayDatePicker: React.FunctionComponent<IProps> = React.forwardRef<
 			normalizeDate(initialMonth)
 		);
 
+		// TODO - Rename to daysSelected
+		const [daysSelected, setDaysSelected] = React.useState<TDaysSelected>(
```

Comment:
Weird that TS can't infer the type here...

Reply:
I created `TDaysSelected` just to have a better name than <[Date, Date]>. So, isn't a good practice?

Reply:
I wasn't thinking about best practices or anything; just surprised that TS didn't infer it automatically (because it is usually pretty good at figuring out the types involved in a `useState` call). I mean, you might think it could be confused by the `Object.keys().reduce()` stuff in `setDate()` (which is called from `normalizeDate()`), and `setDate()` doesn't have a return type annotation... but my editor says TS _is_ inferring it `setDate` as:

```
function setDate(...blah}): Date
```

and the return type (`Date`) is correct.

Just testing this locally, I see that if you remove the explicit annotation the type of the `useState` gets inferred as `Date[]` as opposed to `[Date, Date]` (ie. array instead of tuple), so what you have here is stricter, and stricter is good. Personally though, I would just give TS a hint (`as const`) that you want a tuple here, and drop the annotation:

```ts
const [daysSelected, setDaysSelected] = React.useState(
	() => [normalizeDate(initialMonth), normalizeDate(initialMonth)] as const
);
```

Same with the `useState<Date>` a few lines above; if you drop the `<Date>` it can still figure out the type:

```ts
const [currentMonth, setCurrentMonth] = React.useState(() =>
	normalizeDate(initialMonth)
);
```

My rationale here is that our plans to adopt TS in liferay-portal are going to face less resistance from people unfamiliar with TS if we avoid the syntax and clutter of type annotations in places that they can be inferred (the type info is always only a tool-tip away, anyway).

Separate topic is whether:

- `TDaysSelected`
- `[Date, Date]`

is better or more readable. For simple types, I prefer to "inline" them because I can tell immediately _exactly_ what `[Date, Date]` is, whereas with `TDaysSelected` I have to do some guess-work: is it an object with day keys and `true`/`false` values? An array with one slot for each day of the week? One slot for each day of the month? Is it a `Set` containing a bunch of `Date` objects? etc.

Reply:
I didn't knew about `as const` for tuples. Thanks for helping :)))

About the `TDaysSelected` makes sense don't use a separated type definition on simpler types. I'm removing and replacing with `readonly [Date, Date]` ;)